From 6d767b755ff896bcca2f5a1c273441f7677b4b42 Mon Sep 17 00:00:00 2001
From: Ricardo Salveti <ricardo@foundries.io>
Date: Tue, 14 Apr 2020 20:09:48 -0300
Subject: [PATCH] Distinguish XSI-compliant strerror_r

XSI-compliant strerror_r and GNU-specific strerror_r are different.

 /* XSI-compliant */
 int strerror_r(int errnum, char *buf, size_t buflen);

 /* GNU-specific */
 char *strerror_r(int errnum, char *buf, size_t buflen);

We need to distinguish between them otherwise we'll get an int value
assigned to (char *) variable, resulting in seg fault.

Upstream-Status: Inappropriate [musl specific]

Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
---
 shared/nm-glib-aux/nm-errno.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/shared/nm-glib-aux/nm-errno.c b/shared/nm-glib-aux/nm-errno.c
index 30eb9a8..b25aed6 100644
--- a/shared/nm-glib-aux/nm-errno.c
+++ b/shared/nm-glib-aux/nm-errno.c
@@ -121,7 +121,7 @@ nm_strerror_native_r (int errsv, char *buf, gsize buf_size)
 	nm_assert (buf);
 	nm_assert (buf_size > 0);
 
-#if (_POSIX_C_SOURCE >= 200112L) && !  _GNU_SOURCE
+#ifndef __GLIBC__
 	/* XSI-compliant */
 	{
 		int errno_saved = errno;
-- 
2.26.0

